Signals/DataAccess/Repositories/SignalsDataRepository.cs

```diff
@@ -49,12 +49,23 @@ public IEnumerable<Datum<T>> GetData<T>(Signal signal, DateTime fromIncludedUtc,
         public void SetData<T>(IEnumerable<Datum<T>> data)
         {
             var concreteDatumType = GetConcreteDatumType<T>();
+            var oldData = data.Count() > 0
```

Comment:
`oldData` -> `existingData` perhaps (consistent with naming in `SetMissingValuePolicy`)?